File: ByC/二分查找.c
#include <stdio.h>

int search(int* nums, int numsSize, int target) {
    int left = 0;
    int right = numsSize - 1;
    while (left <= right) {
        int i = left + (right - left) / 2;
        if (nums[i] == target)
            return i;
        else if (nums[i] < target)
            left = i + 1;
        else
            right = i - 1;
    }
    return -1;
}

File: Bypyhton/二分查找.py


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        while left <= right:
            i = int((left + right) / 2)
            if nums[i] < target:
                left = i + 1
            elif nums[i] > target:
                right = i - 1
            else:
                return i
        return -1